lib/Lib.ByteSequence.fst

```diff
@@ -834,3 +834,40 @@ let rec nat_from_intseq_le_public_to_secret #t len b =
     nat_from_intseq_le_lemma0 (slice b_secret 0 1);
     eq_intro (slice (map secret b) 1 len) (map secret (slice b 1 len));
     () end
+
+
+let lemma_uints_to_bytes_le_sub #t #l #len s i =
```

Comment:
You could simplify a `lemma_uints_to_bytes_le_i_j`lemma by having only one last argument:
```
let lemma_uints_to_bytes_le_sub #t #l #len s i =
  let lemma_uints_to_bytes_le_i_j (j : size_nat {j < numbytes t}):
      Lemma(index (uints_to_bytes_le #t #l s) (i * numbytes t + j) == index (uint_to_bytes_le (index s i)) j) =
    index_uints_to_bytes_le #t #l #len s (i*numbytes t + j);
    assert (index (uints_to_bytes_le #t #l s) (i*numbytes t + j) ==
           index (uint_to_bytes_le (index s i)) j) in

  Classical.forall_intro lemma_uints_to_bytes_le_i_j;
  eq_intro (sub (uints_to_bytes_le #t #l s) (i * numbytes t) (numbytes t)) (uint_to_bytes_le (index s i))
```

Reply:
Done